src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -107,6 +115,10 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     var platform =
         CloudPlatformWrapper.of(sourceDataset.getDatasetSummary().getStorageCloudPlatform());
 
+    // Take out a shared lock on the source dataset, to guard against it being deleted out from
+    // under us (for example)
+    addStep(new LockDatasetStep(datasetService, datasetId, true), randomBackoffRetry);
```

Comment:
Note for reviewers:

Awhile back, snapshots would take out exclusive locks on their source datasets on creation, but this was overkill and caused issues for reasonable concurrent dataset operations.  The lock was [fully removed](https://github.com/DataBiosphere/jade-data-repo/pull/1094) but a more appropriate move is to instead take out a shared lock.

Reply:
Added unit tests to verify this behavior.